File: Lesson12_Inheritance/06_Books/BooksTest.py
"""
Copyright Khawla Shnaikat and ChatGPT 3.5, 2023
Licensed under GPL v3
See LICENSE.txt for more information.
To run these tests: 
- Open a terminal or command prompt.
- Navigate to the directory where test_books.py and books.py are saved on your device.
- Run the tests by typing python -m unittest test_books.py.
"""
import unittest
from books import Anthology, Novel, Nonfiction

class BooksTest(unittest.TestCase):

    def test_check_relationship_anthology_cover_art(self):
        anthology = Anthology()
        expected = "Method coverArt called from Anthology"
        self.assertEqual(expected, anthology.cover_art())

    def test_anthology_default_constructor(self):
        new_anthology = Anthology()
        self.assertIsNotNone(new_anthology)

    def test_anthology_inherited_constructor(self):
        isbn = "0987654321"
        pages = 10
        new_anthology = Anthology(isbn, pages)
        self.assertIsNotNone(new_anthology)

    def test_paperback_cover_art(self):
        new_nf = Nonfiction()
        expected = "Method coverArt called from Paperback"
        self.assertEqual(expected, new_nf.cover_art())

    def test_check_relationship_novel_cover_art(self):
        novel = Novel()
        expected = "Method coverArt called from Novel"
        self.assertEqual(expected, novel.cover_art())

    def test_fiction_genre(self):
        antho = Anthology()
        expected = "Method genre called from Fiction"
        self.assertEqual(expected, antho.genre())

    def test_check_relationship_nonfiction_topic(self):
        nonfiction = Nonfiction()
        expected = "Method topic called from Nonfiction"
        self.assertEqual(expected, nonfiction.topic())

if __name__ == '__main__':
    unittest.main()

File: Lesson12_Inheritance/06_Books/BooksTest.java
package edu.ucalgary.oop;

import static org.junit.jupiter.api.Assertions.fail;
import org.junit.Test;
import org.junit.Before;
import org.junit.Assert;
import static org.junit.Assume.*;
import static org.junit.Assert.*;
import java.util.Arrays;
import org.junit.jupiter.api.extension.ExtendWith;
import jdk.jfr.Timestamp;
import java.lang.reflect.Method;
import org.junit.Rule;
import org.junit.rules.ExpectedException;


/*
The following are a selection of tests to get you started with checking your Books.java implementation. Feel free to add your own!
*/


public class BooksTest{
    
    String isbn = "0987654321";
    int pages = 10;

	public BooksTest(){
	}

    @Test
    public void checkRelationship_AnthologyCoverArt(){
       Anthology anthology = new Anthology();
       String expected = "Method coverArt called from Anthology";
       assertEquals("Expected statement was not returned from coverArt() in the Anthology class: ", expected,anthology.coverArt());        
    }

    @Test
    public void testAnthology_DefaultConstructor(){
        
        Anthology newAnthology = new Anthology();
        assertNotNull("Default constructor failed, new Anthology object is null: ", newAnthology);
    }

    @Test
    public void testAnthology_InheritedConstructor(){
        
        Anthology newAnthology = new Anthology(isbn, pages);
        assertNotNull("Inherited constructor failed, new Anthology object is null: ", newAnthology);
    }

    @Test
    public void testPaperbackCoverArt(){
     
        Nonfiction newNF = new Nonfiction();
		String expected = "Method coverArt called from Paperback";
        assertEquals("Expected statement was not returned from coverArt() in the Paperback class: ", expected, newNF.coverArt());
        
    }

    @Test
    public void checkRelationship_NovelCoverArt(){
       Novel novel = new Novel();
       String expected = "Method coverArt called from Novel";
       assertEquals("Expected statement was not returned from coverArt() in the Novel class: ", expected,novel.coverArt());        
    }
	

    @Test
    public void testFictionGenre(){
     
        Anthology antho = new Anthology();
        String expected = "Method genre called from Fiction";
        assertEquals("Expected statement was not returned from genre() in the Fiction class: ", expected,antho.genre());
    }
	
    @Test
    public void checkRelationship_NonfictionTopic(){
        Nonfiction nonfiction = new Nonfiction();
        String expected = "Method topic called from Nonfiction";
        assertEquals("Expected statement was not returned from topic() in the Nonfiction class: ", expected,nonfiction.topic());        
    }

	

}
